src/Avalonia.X11/X11TrayIconImpl.cs

```diff
@@ -0,0 +1,374 @@
+﻿using System;
+using System.Diagnostics;
+using System.Reactive.Disposables;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Avalonia.Controls.Platform;
+using Avalonia.FreeDesktop;
+using Avalonia.Platform;
+using Tmds.DBus;
+
+[assembly: InternalsVisibleTo(Tmds.DBus.Connection.DynamicAssemblyName)]
+
+namespace Avalonia.X11
+{
+    internal class X11TrayIconImpl : ITrayIconImpl
+    {
+        
+        private static int s_trayIconInstanceId = 0;
+        private static int GetTID() => s_trayIconInstanceId++;
+        private ObjectPath _dbusmenuPath;
+        private StatusNotifierItemDbusObj _statusNotifierItemDbusObj;
+        private Connection con;
+        private DbusPixmap _icon;
+        
+        private IStatusNotifierWatcher _statusNotifierWatcher;
+
+        private string _sysTrayServiceName;
+        private string _tooltipText;
+        private bool _isActive;
+        private bool _isDisposed;
+        private readonly bool _ctorFinished;
+        
+        public INativeMenuExporter MenuExporter { get; }
+        public Action OnClicked { get; set; }
+        
+        public X11TrayIconImpl()
+        {
+            con = DBusHelper.TryGetConnection();
```

Comment:
Note that DBus is not guaranteed to be available. We need to properly handle that.

Reply:
@jmacato

Reply:
this is handled.